### PR TITLE
fix: problem with multibyte characters in julia engine

### PR DESCRIPTION
**Pull request by MHellmund**
## Description

This concerns the new julia engine and solves issue https://github.com/PumasAI/QuartoNotebookRunner.jl/issues/130

## Checklist

I have (if applicable):

- [x] filed a [contributor agreement](https://github.com/quarto-dev/quarto-cli/blob/main/CONTRIBUTING.md).
- [x] referenced the GitHub issue this PR closes
- [ ] updated the appropriate changelog

### CODE REVIEW

File: src/execute/julia.ts [hunk 1]
```diff
@@ -600,16 +600,21 @@ async function writeJuliaCommand(
...
+    let response = "";
+    const newlineAt = restOfPreviousResponse.indexOf(10);
     // if we already have a newline, we don't need to read from conn
-    if (newlineAt !== -1) {
-      restOfPreviousResponse = response.substring(newlineAt + 1);
-      response = response.substring(0, newlineAt);
+    if (newlineAt !== -1 && newlineAt < restLength ) {
+      response = new TextDecoder().decode(restOfPreviousResponse.slice(0, newlineAt));
+      restOfPreviousResponse.set(restOfPreviousResponse.slice(newlineAt + 1, restLength));
+      restLength -= newlineAt+1;
```
Comment (cscheid) [inline src/execute/julia.ts:614]: formatting issue here `newlineAt+1` -> `newlineAt + 1`.

We autoformat the code in VS code; could you run `deno fmt julia.ts` on your side? (if you don't have `deno` installed on your system, you can grab the one that Quarto installs at `package/dist/bin/tools/deno` in your quarto-cli repo)

File: src/execute/julia.ts [hunk 2]
```diff
@@ -600,16 +600,21 @@ async function writeJuliaCommand(
...
-  let restOfPreviousResponse = "";
+  let restOfPreviousResponse = new Uint8Array(512);
+  let restLength = 0; // number of valid bytes in restOfPreviousResponse
   while (true) {
-    let response = restOfPreviousResponse;
-    const newlineAt = response.indexOf("\n");
+    const respArray : Uint8Array[] = [];
+    let respLength = 0;
+    let response = "";
+    const newlineAt = restOfPreviousResponse.indexOf(10);
```
Comment (cscheid) [inline src/execute/julia.ts:609]: (Leaving a comment here for someone else who might wonder)

Searching for a naked `10` value in a naked UTF-8 stream looks a bit suspicious, but actually does work. The UTF-8 encoding is clever in that all extended code units appear in the byte stream as bytes that have `10` in [their most significant two bits](https://en.wikipedia.org/wiki/UTF-8#Encoding). Nice!

**Review by cscheid — CHANGES_REQUESTED** [review]
I'd like to ensure consistent formatting, but the content looks good. Thank you!

**Comment by cscheid** [comment]
@MHellmund We also need a contributor agreement from you.

File: src/execute/julia.ts (hunk 1 above)
Comment (MHellmund) [reply src/execute/julia.ts]: ok, I did run `deno fmt`

**Review by cscheid — APPROVED** [review]

**Comment by MHellmund** [comment]
I just sent the contributor agreement to jj@rstudio.com

**Comment by MHellmund** [comment]
Please don't merge yet, I think I found a problem.

**Comment by cscheid** [comment]
Got it. Thanks for following up, and good thing I was out on vacation most of last week! :)

**Comment by MHellmund** [comment]
It's amazing how many tests a wrong code can pass before  you run into heisenbugs with a larger document.
I think I've got it now.

**Comment by cscheid** [comment]
Thanks, @MHellmund! If you have tests you can share publicly, we'd love to run them in our suite going forward.

**Comment by MHellmund** [comment]
There is the test enclosed to the original bug report https://github.com/PumasAI/QuartoNotebookRunner.jl/issues/130 .
[This](https://www.math.uni-leipzig.de/~hellmund/juliabook/)  Quarto book is produced with the new engine. [Source](https://gitea.math.uni-leipzig.de/hellmund/JuliaKurs23). But I don't have a more formal test.

**Pull request merged**